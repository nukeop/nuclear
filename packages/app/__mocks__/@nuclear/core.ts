const initialStoreState = () => ({
  equalizer: {
    selected: 'Default'
  },
  downloads: [],
  favorites: {
    albums: [],
    tracks: []
  },
  playlists: []
});

let mockStore = initialStoreState();

module.exports = {
  store: {
    get: (key: string) => mockStore[key] || {},
    set: (key: string, value: any) => {
      mockStore[key] = value;
    },
    clear: () => mockStore = initialStoreState()
  },
  createApi: () => ({
    app: {},
    store: {},
    React: jest.fn(),
    ReactDOM: jest.fn()
  }),
  getOption: () => '',
  rest: {
    LastFmApi: class {
      constructor() {}

      getTagInfo() {}
      getTagTracks() {}
      getTagAlbums() {}
      getTagArtists() {}
      getTopTags = () => Promise.resolve()
      getTopTracks = () => Promise.resolve()
    }
  },
  settingsConfig: jest.requireActual('@nuclear/core/src/settings').settingsConfig,
  SettingType: {
    BOOLEAN: 'boolean',
    LIST: 'list',
    NODE: 'node',
    NUMBER: 'number',
    STRING: 'string',
    DIRECTORY: 'directory'
  }
};
