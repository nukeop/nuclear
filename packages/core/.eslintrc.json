{
  "extends": [
    "../../.eslintrc.json"
  ],
  "env": {
    "node": true,
    "browser": true,
    "commonjs": true
  },
  "rules": {
    "node/no-missing-import": 0,
    "node/no-extraneous-import": 0
  }
}
