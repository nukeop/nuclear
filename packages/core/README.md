# `@nuclear/core`

> TODO: description

## Usage

```
const core = require('@nuclear/core');

// TODO: DEMONSTRATE API
```
