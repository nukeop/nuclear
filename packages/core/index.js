export {
  LastFmApi,
  isElectron
} from './lib';
