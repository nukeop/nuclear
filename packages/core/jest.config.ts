export default async () => ({
  transform: {
    '^.+\\.(ts|tsx|js|jsx)$': 'ts-jest'
  },
  moduleNameMapper: {
    '^react$': '<rootDir>/../../node_modules/react'
  }
});
