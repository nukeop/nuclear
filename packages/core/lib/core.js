'use strict';

module.exports = core;

function core() {
  // TODO
}
