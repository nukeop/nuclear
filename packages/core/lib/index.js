import 'isomorphic-fetch';
export { default as LastFmApi } from './rest/lastfm';
