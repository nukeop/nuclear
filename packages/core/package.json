{
  "name": "@nuclear/core",
  "version": "0.6.16",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/nukeop/nuclear.git"
  },
  "author": "nukeop <nukeop@gumblert.tech>",
  "license": "AGPL-3.0",
  "bugs": {
    "url": "https://github.com/nukeop/nuclear/issues"
  },
  "homepage": "https://github.com/nukeop/nuclear#readme",
  "main": "src/index.ts",
  "scripts": {
    "lint": "eslint \"{test,src}/**/**.{js,ts}\"",
    "lint:fix": "eslint \"{test,src}/**/**.{js,ts}\" --fix",
    "test": "jest",
    "coverage": "jest --coverage"
  },
  "dependencies": {
    "@babel/core": "^7.9.0",
    "@babel/preset-env": "^7.9.5",
    "@babel/preset-react": "^7.9.4",
    "babel-jest": "^26.6.3",
    "bandcamp-scraper": "1.4.1",
    "cheerio": "^1.0.0-rc.5",
    "deezer-public-api": "^1.0.4",
    "electron-store": "^5.1.0",
    "electron-timber": "^0.5.1",
    "get-artist-title": "^1.2.0",
    "get-lyrics-hd": "^0.0.1",
    "isomorphic-fetch": "^2.2.1",
    "jest": "^26.6.3",
    "md5": "^2.2.1",
    "megalodon": "^3.6.1",
    "nodebrainz": "^2.1.1",
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    "redux": "^4.0.5",
    "search-azlyrics": "0.0.3",
    "ts-jest": "^26.5.5",
    "youtube-playlist": "^1.0.2",
    "ytsr": "^3.5.0"
  },
  "devDependencies": {
    "@types/isomorphic-fetch": "0.0.35",
    "@types/jest": "^26.0.22",
    "@types/lodash": "^4.14.158",
    "@types/md5": "^2.1.33",
    "@types/react": "^16.9.43",
    "eslint": "^7.18.0",
    "mock-require": "^3.0.3",
    "ts-node": "^8.10.2",
    "typescript": "^4.2.4",
    "ytdl-core": "^4.8.3"
  }
}
