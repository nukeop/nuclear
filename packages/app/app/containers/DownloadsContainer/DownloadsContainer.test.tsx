import { store as electronStore } from '@nuclear/core';
import { waitFor } from '@testing-library/dom';
import { buildStoreState } from '../../../test/storeBuilders';
import { mountedComponentFactory, setupI18Next } from '../../../test/testUtils';

const initialStoreState = 
buildStoreState()
  .withDownloads()
  .withPlugins()
  .withConnectivity()
  .build();

describe('Downloads container', () => {
  beforeAll(() => {
    setupI18Next();
  });

  beforeEach(() => {
    electronStore.set(
      'downloads',
      initialStoreState.downloads
    );
  });

  it('should display downloads', () => {
    const { component } = mountComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should clear finished downloads', async () => {
    const { component } = mountComponent();
    await waitFor(() => component.getByText(/clear finished tracks/i).click());

    expect(component.queryByText(/test artist 1 - finished track/i)).toBeNull();
  });

  it('should remove a track', async () => {
    const { component } = mountComponent();
    await waitFor(() => component.getAllByTestId(/remove-download/i)[0].click());

    expect(component.queryByText(/test artist 1 - finished track/i)).toBeNull();
  });

  it('should pause a download in progress', async () => {
    const { component, store } = mountComponent();
    await waitFor(() => component.getAllByTestId(/download-action/i)[3].click());

    const state = store.getState();
    expect(state.downloads[3].status).toBe('Paused');
  });

  it('should resume a paused download', async () => {
    const { component, store } = mountComponent();
    await waitFor(() => component.getAllByTestId(/download-action/i)[2].click());

    const state = store.getState();
    expect(state.downloads[2].status).toBe('Waiting');
  });

  it('should retry a download with error', async () => {
    const { component, store } = mountComponent();
    await waitFor(() => component.getAllByTestId(/download-action/i)[1].click());

    const state = store.getState();
    expect(state.downloads[1].status).toBe('Waiting');
  });

  it('should set downloads dir', async () => {
    const { component } = mountComponent();
    await waitFor(() => component.getByText(/choose a directory.../i).click());

    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const remote = require('electron').remote;
    expect(remote.dialog.showOpenDialog).toHaveBeenCalledWith({
      properties: ['openDirectory']
    });
  });

  const mountComponent = mountedComponentFactory(
    ['/downloads'],
    initialStoreState
  );
});
