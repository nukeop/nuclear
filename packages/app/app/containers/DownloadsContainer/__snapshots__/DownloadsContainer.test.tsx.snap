// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downloads container should display downloads 1`] = `
<DocumentFragment>
  <div
    class="main_layout_container"
  >
    <div
      class="downloads_container"
    >
      <div
        class="ui segment downloads_header"
      >
        <span
          class="label"
        >
          Saving in:
          <span
            class="directory"
          />
        </span>
        <button
          class="ui icon inverted left labeled button"
        >
          <i
            aria-hidden="true"
            class="folder open icon"
          />
          Choose a directory...
        </button>
      </div>
      <div
        class="header_container"
      >
        Downloads
      </div>
      <div
        class="ui inverted segment"
      >
        <button
          class="ui primary button"
        >
          <i
            aria-hidden="true"
            class="trash icon"
          />
          Clear finished tracks
        </button>
        <table
          class="ui inverted table downloads_list"
        >
          <thead
            class=""
          >
            <tr
              class=""
            >
              <th
                class=""
              >
                Status
              </th>
              <th
                class=""
              >
                Name 
                <i
                  aria-hidden="true"
                  class="caret up icon"
                />
              </th>
              <th
                class=""
              >
                Completion
              </th>
              <th
                class=""
              />
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr
              class="downloads_item"
            >
              <td
                class=""
              >
                <i
                  aria-hidden="true"
                  class="green checkmark icon"
                />
                Finished
              </td>
              <td
                class=""
              >
                test artist 1 - finished track
              </td>
              <td
                class=""
              >
                100%
              </td>
              <td
                class="item_buttons"
              >
                <a
                  data-testid="download-action"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="redo fitted icon"
                  />
                </a>
                <a
                  data-testid="remove-download"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="times fitted icon"
                  />
                </a>
              </td>
            </tr>
            <tr
              class="downloads_item"
            >
              <td
                class=""
              >
                <i
                  aria-hidden="true"
                  class="red times icon"
                />
                Error
              </td>
              <td
                class=""
              >
                test artist 2 - track with errorx
              </td>
              <td
                class=""
              >
                10%
              </td>
              <td
                class="item_buttons"
              >
                <a
                  data-testid="download-action"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="redo fitted icon"
                  />
                </a>
                <a
                  data-testid="remove-download"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="times fitted icon"
                  />
                </a>
              </td>
            </tr>
            <tr
              class="downloads_item"
            >
              <td
                class=""
              >
                <i
                  aria-hidden="true"
                  class="pause circle icon"
                />
                Paused
              </td>
              <td
                class=""
              >
                test artist 3 - paused track
              </td>
              <td
                class=""
              >
                30%
              </td>
              <td
                class="item_buttons"
              >
                <a
                  data-testid="download-action"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="play fitted icon"
                  />
                </a>
                <a
                  data-testid="remove-download"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="times fitted icon"
                  />
                </a>
              </td>
            </tr>
            <tr
              class="downloads_item"
            >
              <td
                class=""
              >
                <i
                  aria-hidden="true"
                  class="cloud download icon"
                />
                Started
              </td>
              <td
                class=""
              >
                test artist 4 - started track
              </td>
              <td
                class=""
              >
                50%
              </td>
              <td
                class="item_buttons"
              >
                <a
                  data-testid="download-action"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="pause fitted icon"
                  />
                </a>
                <a
                  data-testid="remove-download"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="times fitted icon"
                  />
                </a>
              </td>
            </tr>
            <tr
              class="downloads_item"
            >
              <td
                class=""
              >
                <i
                  aria-hidden="true"
                  class="hourglass start icon"
                />
                Waiting
              </td>
              <td
                class=""
              >
                test artist 5 - waiting track
              </td>
              <td
                class=""
              >
                0%
              </td>
              <td
                class="item_buttons"
              >
                <a
                  data-testid="download-action"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="pause fitted icon"
                  />
                </a>
                <a
                  data-testid="remove-download"
                  href="#"
                >
                  <i
                    aria-hidden="true"
                    class="times fitted icon"
                  />
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
